sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/ActionDescriptorFactory.scala

```diff
@@ -69,39 +74,84 @@ private[impl] object ActionDescriptorFactory extends ComponentDescriptorFactory
           kalix.MethodOptions.newBuilder().setEventing(subscriptionOptions).build()
 
+        if (hasEventSourcedEntitySubscription(component) &&
+          findEventSourcedEntityType(method) == findEventSourcedEntityType(component))
+          throw InvalidComponentException(
+            s"method [$method] and class [$component] are subscribed to the same event sourced entity.")
+
         KalixMethod(SubscriptionServiceMethod(method))
           .withKalixOptions(kalixOptions)
       }
 
+    val subscriptionEventSourcedEntityClass =
+      if (hasEventSourcedEntitySubscription(component)) {
+        component.getMethods.sorted // make sure we get the methods in deterministic order
+          .filter(hasActionOutput)
+          .collect {
+            case method
+                if !subscriptionEventSourcedEntityMethods.exists(s =>
+                  s.serviceMethod.methodName == method.getName) => // individual annotated overrides class level annotation
+              val subscriptionOptions = eventingInForEventSourcedEntity(component)
+              val kalixOptions =
+                kalix.MethodOptions.newBuilder().setEventing(subscriptionOptions).build()
+
+              KalixMethod(SubscriptionServiceMethod(method))
+                .withKalixOptions(kalixOptions)
+          }
+      } else Array.empty[KalixMethod]
+
     val subscriptionTopicMethods = component.getMethods
       .filter(hasTopicSubscription)
       .sorted // make sure we get the methods in deterministic order
       .map { method =>
         val subscriptionOptions = eventingInForTopic(method)
         val kalixOptions =
           kalix.MethodOptions.newBuilder().setEventing(subscriptionOptions).build()
+        if (hasTopicSubscription(component)
+          && findSubscriptionTopicName(method) == findSubscriptionTopicName(component))
+          throw InvalidComponentException(s"method [$method] and class [$component] are subscribe to the same topic.")
 
         KalixMethod(SubscriptionServiceMethod(method))
           .withKalixOptions(kalixOptions)
       }
 
-    def combineByTopic(subscriptions: Seq[KalixMethod]): Seq[KalixMethod] = {
+    val subscriptionTopicClass: Array[KalixMethod] =
+      if (hasTopicSubscription(component)) {
+        component.getMethods.sorted // make sure we get the methods in deterministic order
+          .filter(hasActionOutput)
+          .collect {
+            case method
+                if !subscriptionTopicMethods.exists(s =>
+                  s.serviceMethod.methodName == method.getName) => // individual annotated overrides class level annotation
+              val subscriptionOptions = eventingInForTopic(component)
+              val kalixOptions =
+                kalix.MethodOptions.newBuilder().setEventing(subscriptionOptions).build()
+
+              KalixMethod(SubscriptionServiceMethod(method))
+                .withKalixOptions(kalixOptions)
+          }
+      } else Array.empty[KalixMethod]
```

Comment:
If I'm understanding it correctly, it's possible to mix type level annotations with method level annotations as long as they are not on the same topic/entity. Is that correct? 

Do we also support it in the gRPC SDK? 

Because if that's the case, we are introducing a bug here. If we set the type level one to `ignoreUnknown` that will also affect the individually annotated methods.

Reply:
I think this is more than a bug. We are just checking for the type level `ignore` in the router. We should not allow different topic/entity then. This is a problem.

Reply:
It turns out that the protocol first doesn't allow both annotations. Type and method at the same time. I'll follow that path.

Reply:
done